### PR TITLE
Add tests for range excluded bounds of first and last element

### PR DESCRIPTION
**Pull request by fpieper**

### CODE REVIEW

**Comment by flix59** [comment]
- Fixed bug in the avltree:
  - within_bounds calculated if somethign is outside bounds
  - most things worked because it was adapted at the reversed within_bounds
  - it broke at the edge cases, because there smaller equalls should have been equals.
  - reversed within_bounds and adjusted everything around
  - end_bound was not checked for the start node of the range, resulting in invalid iterators
- added tests for edge casese in the iterator

File: integration_tests/tests/test_delete.rs [hunk 1]
```diff
@@ -111,6 +111,23 @@ mod avltree_delete {
...
+        // Tree after inserting:
+        //         48
+        //     27       81
+        // 13   40    72
+        //     35
+        // Tree after deleting 48
+        //         40
+        //     27       81
+        // 13   35    72
+        // Orphant 35 has to find a new place at the position of 40
```
Comment (fpieper) [inline integration_tests/tests/test_delete.rs:125]: Outdated comment?

File: src/avl_tree.rs [hunk 2]
```diff
@@ -1386,27 +1404,30 @@ impl<'a, K: ScryptoSbor + Clone + Ord + Eq + Display + Debug, V: ScryptoSbor + C
     }
 }
 
+
 trait WithinBound<K> {
     fn within_bound(&self, key: &K, direction: Direction) -> bool;
 }
 
-impl<K: Ord> WithinBound<K> for Bound<&K> {
+impl<K: Ord + Debug> WithinBound<K> for Bound<&K> {
```
Comment (fpieper) [inline src/avl_tree.rs:1412]: remove Debug

File: integration_tests/tests/test_insert.rs [hunk 3]
```diff
@@ -151,4 +151,21 @@ mod avltree_insert {
...
         helper.get_range_success(i32::MIN, i32::MAX, output_expected, true);
     }
+    #[test]
+    fn test_insert_gives_correct_return_value() {
+        let mut helper = TestHelper::new();
+        helper.instantiate_default(false);
+        helper.insert(1, 1);
+        let receipt = helper.execute_expect_success(false);
+        let insert_result: Vec<Option<i32>> = receipt.outputs("insert");
+        assert_eq!(insert_result[0], None);
```
Comment (fpieper) [inline integration_tests/tests/test_insert.rs:161]: use vec![] in the asserts (is the more complete check)
